File: aio/chimera-II.py
chimin = open("chimin.txt", "r")
chimout = open("chimout.txt", "w")
 
length = int(chimin.readline())
word1 = chimin.readline().strip()
word2 = chimin.readline().strip()
target = chimin.readline().strip()
 
possible = True
splices = 0
same = False
 
for n in range(length):
    if n == 0 or same:
        if word1[n] == word2[n] and word1[n] == target[n]:
            same = True
        elif word1[n] == target[n]:
            same = False
            last = word1
        elif word2[n] == target[n]:
            same = False
            last = word2
        else:
            possible = False
            break
    else:
        if last[n] == target[n]:
            continue
        else:
            if word1[n] == target[n]:
                splices += 1
                last = word1
            elif word2[n] == target[n]:
                splices += 1
                last = word2
            else:
                possible = False
                break
 
if possible:
    chimout.write("SUCCESS\n")
    chimout.write(str(splices))
else:
    chimout.write("PLAN FOILED")

File: aio/chimera-II.cpp
#include <iostream>
#include <cstdio>
#include <string>

using namespace std;

int main() {
    freopen("chimin.txt", "r", stdin);
    freopen("chimout.txt", "w", stdout);

    int n;
    string a;
    string b;
    string splice;
    int splicesa = 0;
    int splicesb = 0;

    cin >> n >> a >> b >> splice;

    int current = 0;

    for (int i=0; i < n; i++) {
        if (current == 0) {
            if (a[i] != splice[i]) {
                if (b[i] != splice[i]) {
                    splicesa = -1;
                    break;
                }
                
                else {
                    splicesa ++;
                    current = 1;
                }
            }
        }

        else {
            if (b[i] != splice[i]) {
                if (a[i] != splice[i]) {
                    splicesa = -1;
                    break;
                }

                else {
                    splicesa ++;
                    current = 0;
                }
            }
        }
    }

    current = 1;

    for (int i=0; i < n; i++) {
        if (current == 0) {
            if (a[i] != splice[i]) {
                if (b[i] != splice[i]) {
                    splicesb = -1;
                    break;
                }
                
                else {
                    splicesb ++;
                    current = 1;
                }
            }
        }

        else {
            if (b[i] != splice[i]) {
                if (a[i] != splice[i]) {
                    splicesb = -1;
                    break;
                }

                else {
                    splicesb ++;
                    current = 0;
                }
            }
        }
    }

    if (splicesa == -1) {
        cout << "PLAN FOILED";
    }

    else {
        cout << "SUCCESS\n" << min(splicesa, splicesb);
    }

    return 0;

}
